### DISCUSSION TITLE
Elastic Defend Integration

### DISCUSSION THREAD

**Discussion by EchoGangster** [Security-Onion-Solutions/securityonion#11084] [2.4] [answered]
I have tried to install the Elastic Defend integration a few different times but every time I do the in Fleet the Agent goes unhealthy. The cause of the unhealthy agent is the Elastic Defend integration. Error states 

"Policy response failure

The Endpoint did not apply the Policy correctly. Expand the Policy response above for more details."

There isn't any other information in expanding the Policy Response.
The only other integration currently is, System.

This is on an M1 MAC utilizing the so-elastic-darwin-arm agent download.

**Comment by defensivedepth** [comment]
What do you mean that you are installing the Elastic Defend integration? It is installed and enabled by default for the Policy that is used for the so-elastic-darwin-arm installer.

**Comment by EchoGangster** [comment]
I mean I was adding it to a policy I created.  When I added it to the Policy is when I get the error.

**Reply by defensivedepth** [reply]
Does it work using the existing policy?

**Reply by EchoGangster** [reply]
It does not.  I tried that as well.  Using the default, endpoints_initial policy.

**Reply by defensivedepth** [reply]
There are some additional config needed for certain versions of macOS, see Elastic docs here: https://www.elastic.co/guide/en/security/8.9/elastic-endpoint-deploy-reqs.html

It is on the list to document this on our side.

**Comment by EchoGangster** [comment]
Thanks for the feedback.  I am happy to supply any additional information needed to assist.

**Reply by senatesan** [reply]
I uninstalled the Defend integration then added the transform role to my search nodes as explained in [#10349](https://github.com/Security-Onion-Solutions/securityonion/discussions/10349) and then reinstalled the Defend integration and it fixed my issue.

**Comment by defensivedepth** [comment, marked as answer]
@EchoGangster 

```
Remove the Elastic Defend integration by navigating to Elastic Fleet -> Agent Policies -> endpoints-initial -> Actions -> Delete integration. Then run the following on the Manager:

sudo rm -f /opt/so/state/eaintegrations.txt
sudo so-elastic-fleet-integration-policy-load

If you are running a distributed grid with a Manager or ManagerSearch, you may also need to add the following role to Elasticsearch:
'transform'
sudo vi +24 /opt/so/saltstack/default/salt/elasticsearch/config.map.jinja 
so-elasticsearch-restart
```

**Reply by EchoGangster** [reply]
This did not solve my issue with Elastic Defend on MACos with ARM processor.  I will chalk it up to the MAC at this time.  I do see in Elastic Security that there is an elastic-defend-endpoints policy created by so-elastic.  When you click on it states: "Cannot read properties of undefined (reading 'value')"

**Reply by defensivedepth** [reply]
Internally we currently have Elastic Agent working fine on both Intel & arm macs. 

If `"Cannot read properties of undefined (reading 'value')"` that still shows up, then the fix that I suggested did not get implemented fully.

Are you running a Distributed grid with a dedicated Manager?

**Reply by EchoGangster** [reply]
I am running a Standalone setup.  I have run through the steps again and still getting Unhealthy agent as soon as ED is added.  "Policy response failure"

**Reply by defensivedepth** [reply]
Standalone using the ISO or network setup?

**Reply by EchoGangster** [reply]
Standalone using the iso.

**Comment by defensivedepth** [comment]
The issue sounds like what the hotfix fixed. Can you confirm that you are using the latest version of the ISO? https://blog.securityonion.net/2023/08/security-onion-2410-hotfix-20230821-now.html
